quinn/src/lib.rs

```diff
@@ -672,83 +559,101 @@ pub type Incoming = mpsc::Receiver<NewConnection>;
 
 /// A connection initiated by a remote client.
 pub struct NewConnection {
+    /// The driver for the connection
+    pub driver: ConnectionDriver,
     /// The connection itself.
     pub connection: Connection,
     /// The stream of QUIC streams initiated by the client.
     pub incoming: IncomingStreams,
 }
 
-impl NewConnection {
-    fn new(endpoint: Rc<RefCell<EndpointInner>>, handle: quinn::ConnectionHandle) -> Self {
-        let conn = Rc::new(ConnectionInner {
-            endpoint,
-            handle,
-            side: Side::Server,
-        });
-        NewConnection {
-            connection: Connection(conn.clone()),
-            incoming: IncomingStreams(conn),
-        }
-    }
-}
-
 /// A connection initiated locally.
 pub struct NewClientConnection {
+    /// The driver for the connection
```

Comment:
Worth having a hint that this is a future that should be spawned immediately here.

Reply:
Done.